dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/logging/source/AccessLogSource.java

```diff
@@ -109,31 +109,36 @@ public void process(AccessLogRecordData recordData) {
             HttpResponseMessage response = recordData.getResponse();
 
             if (request != null) {
-                long requestStartTime = recordData.getStartTime();
-                String uriPath = request.getRequestURI();
-                String requestMethod = request.getMethod();
-                String queryString = request.getQueryString();
-                String localIP = recordData.getLocalIP();
-                String localPort = recordData.getLocalPort();
-                String remoteHost = recordData.getRemoteAddress();
-                String userAgent = request.getHeader("User-Agent").asString();
-                String requestProtocol = request.getVersion();
-                long responseSize = recordData.getBytesWritten();
-                int responseCode = response.getStatusCodeAsInt();
-                long elapsedTime = recordData.getElapsedTime();
-                long timestamp = recordData.getTimestamp();
-
-                String sequence = requestStartTime + "_" + String.format("%013X", seq.incrementAndGet());
-
-                AccessLogData data = new AccessLogData(uriPath, requestMethod, queryString, localIP, localPort, requestStartTime, remoteHost, userAgent, requestProtocol, responseSize, responseCode, elapsedTime, timestamp, sequence);
-                bufferMgr.add(data);
+
+                GenericData genData = new GenericData();
+
+                long requestStartTimeVal = recordData.getStartTime();
+                genData.addPair("ibm_requestStartTime", requestStartTimeVal);
+                genData.addPair("ibm_uriPath", request.getRequestURI());
+                genData.addPair("ibm_requestMethod", request.getMethod());
+                genData.addPair("ibm_queryString", request.getQueryString());
+                genData.addPair("ibm_requestHost", recordData.getLocalIP());
+                genData.addPair("ibm_requestPort", recordData.getLocalPort());
+                genData.addPair("ibm_remoteHost", recordData.getRemoteAddress());
+                genData.addPair("ibm_userAgent", request.getHeader("User-Agent").asString());
+                genData.addPair("ibm_requestProtocol", request.getVersion());
+                genData.addPair("ibm_bytesReceived", recordData.getBytesWritten());
+                genData.addPair("ibm_responseCode", response.getStatusCodeAsInt());
+                genData.addPair("ibm_elapsedTime", recordData.getElapsedTime());
+                // Must pass datetime as string, as its value type must be set as string in its kvp
+                genData.addPair("ibm_datetime", Long.toString(recordData.getTimestamp()));
```

Comment:
just add ibm_datetime as a long dateVal... it's formatter's job to decide if we want it to be a formatted like a string.